### DISCUSSION TITLE
Adding custom content before tabs

### DISCUSSION THREAD

**Discussion by kabaljit** [react-native-community/react-native-tab-view#1216] [Q&A]
**Is there a way to add custom content before tabs?** 

### What have you tried

I thought of replacing the current TabBar with a new custom component where I can add the customize content before adding TabBar. But now I stuck because I don't if there is a way to change the tabBar from a screen.

### Screenshots (if applicable)
<img width="538" alt="Screenshot 2020-04-06 at 12 19 58" src="https://user-images.githubusercontent.com/49038559/78555327-b0d25c80-7804-11ea-90f5-78f90f628eda.png">

### Your Environment

| software                     | version
| ---------------------------- | -------
| ios or android               |
| react-native                 |0.61.2
| react-native-tab-view        |2.13.0
| react-native-gesture-handler |
| react-navigation     | 4.3.6
| node                         |
| npm or yarn                  |

**Comment by d41z** [comment]
hi @kabaljit could you solve this?

**Comment by ChintanRajpara** [comment]
you can use this prop

  renderTabBar={(tabBarProps) => (
        <View
          style={{
            height: vertical_48,
            flexDirection: "row",
            width,
          }}
        >
          <View
            style={{
              height: vertical_48,
              aspectRatio: 1,
              backgroundColor: "red",
            }}
          ></View>
          <TabBar
            layout={{ height: vertical_48, width: width - vertical_48 }}
            renderIcon={({}) => {}}
            scrollEnabled
            {...tabBarProps}
            
            style={{
              height: vertical_48,
              width: width - vertical_48,
              backgroundColor: "#000",
              justifyContent: "center",
            }}
       
          />
        </View>
      )}


it worked for me perfectly

**Comment by bobostaykov** [comment]
You can place the TabBar inside a View with a flex of 1. Then you can place any content you like wherever you like, e.g. above the TabBar.
